core/dbt/cli/params.py

```diff
@@ -178,7 +183,7 @@
         "Space-delimited listing of node properties to include as custom keys for JSON output "
         "(e.g. `--output json --output-keys name resource_type description`)"
     ),
-    type=list,
+    type=tuple,
```

Comment:
### Context
When I left this as `type=list`, a couple tests failed:
- https://github.com/dbt-labs/dbt-core/actions/runs/4420754548/

But when I updated it to `type=tuple`, they passed again:
- https://github.com/dbt-labs/dbt-core/actions/runs/4422283702/

Although it is not enforced, there is a comment [here](https://github.com/dbt-labs/dbt-core/blob/311a57a21e61955904202bdb469c2077202e9073/core/dbt/cli/options.py#L5) that says:
> MultiOption options must be specified with type=tuple or type=ChoiceTuple

### Suggestion
To avoid unforeseen issues in the future, maybe we should enforce that the configured `type` is one of the following whenever `cls=MultiOption`?
- subclass of `tuple`
- instance of `ChoiceTuple`

One option for enforcing this within the [`__init__` method of `MultiOption`](https://github.com/dbt-labs/dbt-core/blob/311a57a21e61955904202bdb469c2077202e9073/core/dbt/cli/options.py#L4-L13):
```
option_type = kwargs.pop("type", None)
if inspect.isclass(option_type):
    assert issubclass(option_type, tuple), "type must be tuple or ChoiceTuple not {}".format(option_type)
else:
    assert isinstance(option_type, ChoiceTuple), "type must be tuple or ChoiceTuple not {}".format(option_type)
```